RichEditorView/Classes/RichEditorView.swift

```diff
@@ -441,54 +472,75 @@ import UIKit
     /// For example, if the cursor is directly at the top of what is visible, it will return 0.
     /// This also means that it will be negative if it is above what is currently visible.
     /// Can also return 0 if some sort of error occurs between JS and here.
-    private var relativeCaretYPosition: Int {
-        let string = runJS("RE.getRelativeCaretYPosition();")
-        return Int(string) ?? 0
+    private var relativeCaretYPosition: Int = 0
+    func getRelativeCaretYPosition(onCompletion:@escaping((Int, Error?)-> Void)){
+        webView.evaluateJavaScript("RE.getRelativeCaretYPosition();", completionHandler: { (response: Any?, error: Error?) in
+            if let stringResponse = response as? Int {
+                onCompletion(stringResponse, error)
+            } else {
+                onCompletion(0, error)
+            }
+        })
+        //        runJSX("RE.getRelativeCaretYPosition();") { (response: String, error:Error?) in
+        //            guard error == nil else {onCompletion(0, error); return}
+        //            onCompletion(Int(response) ?? 0, nil)
+        //        }
```

Comment:
If you want to leave commented code, please add comments to explain

Reply:
Leave it there for tdebug, forgot to delete it. It will be removed.